handout/a11y/key-concerns/semantic-markup.md

```diff
@@ -0,0 +1,166 @@
+
+```
+@Component({
+  selector: 'ngc2-app',
+  template: `
+    <ngc2-notification-button
+      message="Hello!"
+      label="Greeting"
+      role="button">
+    </ngc2-notification-button>
+  `,
+  styles: [`
+    div {
+      display: flex;
+      width: 80px;
+      height: 80px;
+      justify-content: center;
+      align-items: center;
+      background-color: yellow;
+      border-radius: 40px;
+    }
+
+    div:hover {
+      cursor: pointer;
+    }
+  `]
+})
+export class AppComponent { }
+
+@Component({
+  selector: 'ngc2-notification-button',
+  template: `
+    <div (click)="notify()">
+      <span>Notify</span>
+    </div>
+  `,
+})
+export class NotificationButtonComponent {
+  @Input()
+  message = 'Alert!';
+
+  @Input()
+  label = 'Notify';
+
+  constructor(private notification: NotificationService) { }
+
+  notify() {
+    this.notification.notify(this.message)
+  }
+}
+```
+[View Example](https://plnkr.co/edit/aAjNnmeaEPbdfIWo9hPT?p=preview)
+
+This lets you create a component that has the same semantics as a `button` element to screen readers and browsers, but now have the opportunity to fully control the styling of that component as well as inject your own custom logic.
+
+
+### ARIA attributes
+
+Some native html tags have attributes that providers extra context on what's being displayed on the browser: The `img` tag's `alt` attribute for example, lets the reader know what is being shown using a short description. However, native tags don't cover all cases. This is where ARIA fits in. ARIA attributes can provide context on what roles specific elements have in the application or can provide context on how some elements are related to other elements on the document. One example of this is modals. Native modals provided by different platforms such as web browsers, often have limited customization options which can make for a poor experience in some cases. This necessitates the creation of custom modals. A modal component can be given the `role` of [dialog](https://www.w3.org/TR/wai-aria/roles#dialog) or [alertdialog](https://www.w3.org/TR/wai-aria/roles#alertdialog) to let the browser know that that component is acting as a modal.  Within the component we can use the ARIA attributes `aria-labelledby` and `aria-described` by to let readers know which elements within those modals say what the title of that modal is and a longer description of what exactly the user is being shown within that modal.
```

Comment:
> Some native html tags have attributes that ~~providers~~ _provide_ extra context on what's being displayed on the browser.

Paragraph break.

> _For example, the_ ~~The~~ `img` tag's `alt` attribute~~ for example,~~ lets the reader know what is being shown using a short description. 

Paragraph break.

> However, native tags don't cover all cases~~. This~~ _, and this_ is where ARIA fits in. 

Paragraph break. 

> ARIA attributes can provide context on what roles specific elements have in the application or can provide context on how some elements are related to other elements ~~on~~ _within_ the document. 

Paragraph break.

> One example of this is modals. 

Paragraph break.

> Native modals provided by different platforms such as web browsers~~,~~ often have limited customization options which_, in some cases,_ can make for a poor experience ~~in some cases.~~ 

Paragraph break.

> This necessitates the creation of custom modals. 

Paragraph break.

> A modal component can be given the `role` of [dialog](https://www.w3.org/TR/wai-aria/roles#dialog) or [alertdialog](https://www.w3.org/TR/wai-aria/roles#alertdialog) to let the browser know that that component is acting as a modal.  

Paragraph break.

> Within the component we can use the ARIA attributes `aria-labelledby` and `aria-described` by to let readers know which elements within those modals say what the title of that modal is and a longer description of what exactly the user is being shown within that modal.

May want to rewrite that last sentence. Reads a little funny.